src/controllers/productController.ts

```diff
@@ -0,0 +1,103 @@
+import { Request, Response } from 'express'
+import cloudinary from 'cloudinary'
+import {
+  CLOUDINARY_CLOUD_NAME,
+  CLOUDINARY_API_KEY,
+  CLOUDINARY_API_SECRET,
+} from '../config/index'
+import Product from '../database/models/productModel'
+import Collection from '../database/models/collectionModel'
+
+cloudinary.v2.config({
+  cloud_name: CLOUDINARY_CLOUD_NAME,
+  api_key: CLOUDINARY_API_KEY,
+  api_secret: CLOUDINARY_API_SECRET,
+})
+
+/**
+ * Product Controller class
+ */
+export default class productController {
+  /**
+   * Adding product
+   * @param {Request} req - Express request object
+   * @param {Response} res - Express response object
+   * @returns {Promise<Response>} Promise that resolves to an Express response
+   */
+  static async addProduct(req: Request, res: Response): Promise<Response> {
+    try {
+      const {
+        name,
+        price,
+        category,
+        collectionId,
+        bonus,
+        sku,
+        quantity,
+        expiryDate,
+      } = req.body
+
+      const sellerId = req.user?.id
+
+      const existingProduct = await Product.findOne({
+        where: { name, sellerId, sku },
+      })
+      if (existingProduct) {
+        return res.status(409).json({
+          message: 'Product exists, please update your stock',
+        })
+      }
+
+      const uploadedImages = await Promise.all(
+        (req.files as Express.Multer.File[]).map(
+          async (file: Express.Multer.File) => {
+            const result = await cloudinary.v2.uploader.upload(file.path)
+            return { secure_url: result.secure_url }
+          },
+        ),
+      )
+
+      const imageUrls = uploadedImages.map((image) => image.secure_url)
+
+      await Product.create({
+        name,
+        price,
+        category,
+        collectionId,
+        bonus,
+        sku,
+        quantity,
+        images: imageUrls,
+        expiryDate,
+        sellerId,
+      })
+
+      return res.status(201).json({ message: 'Product added successfully' })
+    } catch (error) {
+      return res.status(500).json({ message: 'Internal server error' })
+    }
+  }
+
+  /**
+   * Create collection
+   * @param {Request} req - Express request object
+   * @param {Response} res - Express response object
+   * @returns {Promise<Response>} Promise that resolves to an Express response
+   */
+  static async createCollection(
+    req: Request,
+    res: Response,
+  ): Promise<Response> {
+    try {
+      const { name } = req.body
+
+      const collection = await Collection.create({ name })
```

Comment:
Also here you can save the user (userId) who created that collection, this can be helpful.